shipyard_blueprints/s3/shipyard_s3/cli/download.py

```diff
@@ -43,244 +48,116 @@ def get_args():
     return parser.parse_args()
 
 
-def set_environment_variables(args):
+def get_dest_file(
+    src_path: str, target_file: str, file_number: Optional[int] = None
+) -> str:
     """
-    Set AWS credentials as environment variables if they're provided via keyword arguments
-    rather than seeded as environment variables. This will override system defaults.
-    """
-
-    if args.aws_access_key_id:
-        os.environ["AWS_ACCESS_KEY_ID"] = args.aws_access_key_id
-    if args.aws_secret_access_key:
-        os.environ["AWS_SECRET_ACCESS_KEY"] = args.aws_secret_access_key
-    if args.aws_default_region:
-        os.environ["AWS_DEFAULT_REGION"] = args.aws_default_region
-    return
+    Determines the destination file name based on provided parameters.
 
+    Args:
+    source_full_path (str): The full path of the source file.
+    destination_file_name (str): The initial name of the destination file.
+    file_number (int, optional): The number to append to the file name if necessary.
 
-def connect_to_s3(s3_config=None):
+    Returns:
+    str: The determined destination file name.
     """
-    Create a connection to the S3 service using credentials provided as environment variables.
-    """
-    s3_connection = boto3.client("s3", config=Config(s3_config))
-    return s3_connection
+    if not target_file:
+        target_file = os.path.basename(src_path)
+    if file_number:
+        name, ext = os.path.splitext(target_file)
+        target_file = f"{name}_{file_number}{ext}"
+    return target_file
 
 
-def extract_file_name_from_source_full_path(source_full_path):
-    """
-    Use the file name provided in the source_file_name variable. Should be run only
-    if a destination_file_name is not provided.
-    """
-    destination_file_name = os.path.basename(source_full_path)
-    return destination_file_name
+def main():
+    try:
+        args = get_args()
+        bucket_name = args.bucket_name
+        src_file = args.source_file_name
+        src_folder = (
+            shipyard.files.clean_folder_name(args.source_folder_name)
+            if args.source_folder_name
+            else None
+        )
+        match_type = args.source_file_name_match_type
+        target_dir = shipyard.files.clean_folder_name(args.destination_folder_name)
 
+        if not os.path.exists(target_dir) and (target_dir != ""):
```

Comment:
why not use shipyard.files.create_folder_dne
or directly call `os.makedirs(target_dir, exist_ok=True)`